### PR TITLE
6468, 6577, 6578, 6575 misc zfs-tests fixes

### PR DESCRIPTION
**Pull request by yuripankov**
Contains the following issues:

6468 remove iscsi-related cruft from zfs-tests
Remove all shareiscsi leftovers, fixing test case (and removing wildcard characters there as well)

6577 zfs_copies_004_neg fails trying to expand wildcard characters
Remove the wildcard characters from the list of options - they could make the test case fail if eg you have a file named '1' in your home directory, '?' would then expand to '1', which isn't allowed in bad options list

6578 cifs_attr_002_pos fails to cleanup after itself
Fix obvious copy/paste error (missing "\" on multiline statement)

6575 grow_replicas_001_pos fails intermittently
Make the test case run for both mirror/raidz pool types, not randomly selecting one of them (two time during the run, which could make it fail intermittently)

### CODE REVIEW

**Comment by ahrens** [comment]
@zettabot go

**Comment by ahrens** [comment]
@jwk404 can you take a look?

File: usr/src/test/zfs-tests/tests/functional/grow_replicas/Makefile [hunk 1]
```diff
@@ -18,11 +18,9 @@ include $(SRC)/Makefile.master
 ROOTOPTPKG = $(ROOT)/opt/zfs-tests
 TESTDIR = $(ROOTOPTPKG)/tests/functional/grow_replicas
 
-PROGS = cleanup \
-	grow_replicas_001_pos \
-	setup
+PROGS=	grow_replicas_001_pos
```
Comment (jwk404) [inline usr/src/test/zfs-tests/tests/functional/grow_replicas/Makefile]: Please leave spaces around equal sign to match the convention used elsewhere under usr/test.

Comment (yuripankov) [inline usr/src/test/zfs-tests/tests/functional/grow_replicas/Makefile]: OK, changed.

**Comment by ahrens** [comment]
@zettabot go

**Comment by yuripankov** [comment]
Current test results (http://build.prakashsurya.com:8080/job/openzfs-regression-tests/186/) are 404.

**Comment by prakashsurya** [comment]
My bad. We've run out of space on the public mirror:
```
$ df -h /
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1       7.5G  7.2G     0 100% /
```
As a result, the rsync to the public mirror has been failing since the 16th of this month. I'll try and get that addressed tonight (i.e. expand the root device and re-sync).

**Comment by prakashsurya** [comment]
OK, things should be fixed now:
```
$ df -h /var/lib/jenkins
Filesystem      Size  Used Avail Use% Mounted on
/dev/vdb         65G  5.0G   57G   9% /var/lib/jenkins
```
I cleaned things up on the public mirror and pushed this commit to keep us from using the smaller root disk for the jenkins directory: https://github.com/openzfs/openzfs-build/commit/48f561059276e74695d7626856239c1552ca97b2

**Comment by yuripankov** [comment]
Could we re-run the tests, please?

**Comment by ahrens** [comment]
The current test results (http://build.prakashsurya.com:8080/job/openzfs-regression-tests/186/) look good to me -- the failure is in ztest which you haven't changed.  But I'll kick off another run per your request.

**Comment by ahrens** [comment]
@zettabot go

**Comment by ahrens** [comment]
Test failure (ro_props_001_pos) is not related to these changes.

**Pull request closed without merging**